static/js/compose.js

```diff
@@ -795,6 +795,48 @@ exports.render_and_show_preview = function (preview_spinner, preview_content_box
     }
 };
 
+exports.warn_if_private_stream_is_linked = function (linked_stream) {
+    // For PMs, we don't warn about links to private streams, since
+    // you are often specifically encouraging somebody to subscribe
+    // to the stream over PMs.
+    if (compose_state.get_message_type() !== 'stream') {
+        return;
+    }
+
+    const compose_stream = stream_data.get_sub(compose_state.stream_name());
+    if (compose_stream === undefined) {
+        // We have an invalid stream name, don't warn about this here as
+        // we show an error to the user when they try to send the message.
+        return;
+    }
+
+    // If the stream we're linking to is not invite-only, then it's
+    // public, and there is no need to warn about it, since all
+    // users can already see all the public streams.
```

Comment:
This comment is no longer true now that we have guest users; we may want to add a fix-forward to this.